pkg/globalnet/controllers/uninstall.go

```diff
@@ -216,3 +218,38 @@ func RemoveGlobalIPAnnotationOnNode(cfg *rest.Config) {
 
 	logger.Infof("Successfully removed globalIP annotation from node %q", nodeName)
 }
+
+func RemoveStaleInternalServices(config *syncer.ResourceSyncerConfig) error {
+	_, gvr, err := util.ToUnstructuredResource(&corev1.Service{}, config.RestMapper)
+	if err != nil {
+		return errors.Wrap(err, "error converting resource")
+	}
+
+	services := config.SourceClient.Resource(*gvr)
+
+	svcList, err := services.Namespace(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing the services")
+	}
+
+	svc := &corev1.Service{}
+
+	for i := range svcList.Items {
+		if !svcList.Items[i].GetDeletionTimestamp().IsZero() {
+			err = config.Scheme.Convert(&svcList.Items[i], svc, nil)
+			if err != nil {
+				return errors.Wrapf(err, "error converting %#v to Service", svcList.Items[i])
+			}
+
+			if finalizer.IsPresent(&svc.ObjectMeta, InternalServiceFinalizer) {
+				if err := finalizer.Remove(context.TODO(), resource.ForDynamic(services.Namespace(svc.Namespace)), svc,
+					InternalServiceFinalizer); err != nil {
+					return fmt.Errorf("error while removing the finalizer from globalnet internal service %q/%q",
+						svc.Namespace, svc.Name)
+				}
```

Comment:
This loses the details of the error:

```suggestion
				if err := finalizer.Remove(context.TODO(), resource.ForDynamic(services.Namespace(svc.Namespace)), svc,
					InternalServiceFinalizer); err != nil {
					return errors.Wrapf(err, "error while removing the finalizer from globalnet internal service %q/%q",
						svc.Namespace, svc.Name)
				}
```